### PR TITLE
Switch Mqtt Guide to use test resources

### PR DESCRIPTION
**Pull request by timyates**
There were issues with the MQTT guide:

1. The MQTT guide was not using Test Resources (had instructions on setting up a local Mosquitto instance)
2. None of the tests tested MQTT
3. When following the guide, there was no explanation where the "cli" or "app" directories came from.

This PR addresses all these issues.

> 🚨 We uncovered an issue with MQTT detection in Test Resources so until https://github.com/micronaut-projects/micronaut-test-resources/pull/91 is released and available via starter, this PR skips the Maven and Gradle tests.

To run the tests locally, you need to add the HIVEMQ provider manually to the builds generated in `build/code/micronaut-mqtt`

For Gradle:

```
micronaut {
...
    testResources {
...
        additionalModules.add(HIVEMQ)
    }
}
```

For Maven:

```
    <plugins>
      <plugin>
        <groupId>io.micronaut.build</groupId>
        <artifactId>micronaut-maven-plugin</artifactId>
          <configuration>
...
            <testResourcesDependencies>
              <dependency>
                <groupId>io.micronaut.testresources</groupId>
                <artifactId>micronaut-test-resources-hivemq</artifactId>
              </dependency>
            </testResourcesDependencies>
          </configuration>
      </plugin>
```

### CODE REVIEW

**Review by melix — APPROVED** [review]
I wonder if we should simply push this with the Gradle/Maven configuration which works today (that is to say, `additionalModules.add(HIVEMQ)`)

**Comment by timyates** [comment]
> I wonder if we should simply push this with the Gradle/Maven configuration which works today (that is to say, `additionalModules.add(HIVEMQ)`)

Unfortunately, we have no current way of changing the generated builds.  The builds are just created from starter, and then the code is copied across to the src folder.  Not saying we couldn't write some mechanism, but there isn't one there at the moment

**Comment by sdelamo** [comment]
@timyates can we keep this guide to use Mosquitto ?

File: guides/micronaut-mqtt/micronaut-mqtt.adoc [hunk 1]
```diff
@@ -4,23 +4,7 @@ common:requirements.adoc[]
...
-1646389659: mosquitto version 2.0.14 starting
-1646389659: Config loaded from /usr/local/etc/mosquitto/mosquitto.conf.
-1646389659: Starting in local only mode. Connections will only be possible from clients running on this machine.
-1646389659: Create a configuration file which defines a listener to allow remote access.
-1646389659: For more details see https://mosquitto.org/documentation/authentication-methods/
-1646389659: Opening ipv4 listen socket on port 1883.
-1646389659: Opening ipv6 listen socket on port 1883.
-1646389659: mosquitto version 2.0.14 running
-----
+common:test-resources-hivemq.adoc[]
```
Comment (sdelamo) [inline guides/micronaut-mqtt/micronaut-mqtt.adoc]: @timyates  should this be mosquitto again?

Comment (timyates) [reply guides/micronaut-mqtt/micronaut-mqtt.adoc]: Yeah, done in https://github.com/micronaut-projects/micronaut-guides/pull/1064/commits/7ef54ad2d651073cc4bbdda998ea05893530cbbe

**Pull request merged**